IO/logging.conf

```diff
@@ -0,0 +1,10 @@
+[formats]
+console = "%d(%X) [%c:%V] %m%n"
+full = "%d(%F %T) %c:%V [%p:%f:%L] %m%n"
+norms = "%m%n"
+
+[rules]
+*.* >stdout; console
```

Comment:
This `logging.conf` file is copied to each problem directory and you can edit it to control the logging output. You can find the documentation here: http://hardysimpson.github.io/zlog/UsersGuide-EN.html

In short, the `*.*` means _apply to any category and any log level_, so you could change it to `*.INFO`, for instance, to send only messages with a severity of `info` or higher to stdout, which would suppress all `debug` messages. The `console` references a format specified above. Logging to files is set up in the lines below here.

In the code, whenever you would usually call `printf`, you can now instead include `<zlog.h>` and call `zlog_debug`, `zlog_info`, `zlog_error` and so on, depending on the severity you'd like to assign to the message. So for instance:

```c
// Fetch a pointer to some category, you can pass this around or fetch it whenever you need it.
// I suggest using a common variable name such as `c_default` for the appropriate category of the current scope, and different names for specialised categories.
zlog_category_t *c_default = zlog_get_category("d4est_amr");
// Call the log function corresponding to the severity of the event
zlog_info(c_norms, "Performing AMR level %d.", level);
```

I suggest we gradually transition `printf` calls to `zlog` moving forward.

Reply:
This is so much better than the printfs. Thank you for the addition!